File: frontend/src/app/models/car.ts
import {Brand} from './brand';
import {Model} from './model';

export interface Car {
  id: number;
  brand: Brand;
  model: Model;
  description: string;
  productionYear: number;
  hp: number;
  fuel: string;
  transmission: string;
  type: string;
  price: number;
  image: string;
}

File: backend/src/main/java/com/carshop/entities/Car.java
package com.carshop.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Car {

    @Id
    @GeneratedValue
    private Integer id;

    @ManyToOne
    private Brand brand;

    @ManyToOne
    private Model model;

    @Column(columnDefinition = "TEXT")
    private String description;

    private int productionYear;

    private int hp;

    private String fuel;

    private String transmission;

    private String type;

    private double price;

    private String image;

}
